tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -0,0 +1,518 @@
+	// A client for EC2.
+	EC2 *ec2.EC2
+	// A client for IAM.
+	IAM *iam.IAM
+	// openShiftClient is the client of the existing OpenShift cluster.
+	openShiftClient *client.OpenShift
+	// resourceTrackerDir is where `windows-node-installer.json` file is stored.
+	resourceTrackerDir string
+}
+
+// New returns the AWS implementations of the Cloud interface.
+// credentialPath is a directory information to AWS credentials file.
+// credentialAccountID is the account name the user uses to create VM instance.
+// resourceTrackerDir is where created instance and security group information is stored.
+func New(openShiftClient *client.OpenShift, credentialPath, credentialAccountID,
+	resourceTrackerDir string) (*awsProvider, error) {
+	provider, err := openShiftClient.GetCloudProvider()
+	if err != nil {
+		return nil, err
+	}
+	session, err := newSession(credentialPath, credentialAccountID, provider.AWS.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &awsProvider{ec2.New(session, aws.NewConfig()),
+		iam.New(session, aws.NewConfig()),
+		openShiftClient,
+		resourceTrackerDir,
+	}, nil
+}
+
+// newSession uses AWS credentials to create and returns a session for interacting with EC2.
+func newSession(credentialPath, credentialAccountID, region string) (*awssession.Session, error) {
+	if _, err := os.Stat(credentialPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("failed to find AWS credentials from path '%v'", credentialPath)
+	}
+	session := awssession.Must(awssession.NewSession(&aws.Config{
+		Credentials: credentials.NewSharedCredentials(credentialPath, credentialAccountID),
+		Region:      aws.String(region),
+	}))
+	return session, nil
+}
+
+// CreateWindowsVM takes in imageId, instanceType, and sshKey name to create a Windows instance under the same VPC as
+// the existing OpenShift cluster with the following:
+// - attaches existing cloud-specific cluster worker security group and IAM to gain the same access as the linux
+// workers,
+// - uses public subnet,
+// - attaches public ip to allow external access,
+// - adds a security group that allows traffic from within the VPC range and RDP access from user's IP,
+// - uses given image id, instance type, and sshKey name
+// - creates a unique name tag for the instance using the same prefix as the OpenShift cluster name, and
+// - logs id and security group information of the created instance in 'windows-node-installer.json' file at the
+// resourceTrackerDir.
+// On success, the function outputs RDP access information in the commandline interface.
+func (a *awsProvider) CreateWindowsVM(imageId, instanceType, sshKey string) error {
+	infraID, err := a.openShiftClient.GetInfrastructureID()
+	if err != nil {
+		return err
+	}
+	vpcID, err := a.getVPCIdByInfrastructure(infraID)
+	if err != nil {
+		return fmt.Errorf("failed to find cluster VPC, %v", err)
+	}
+	err = a.ensureDnsOnVpcEnabled(vpcID)
+	if err != nil {
+		log.Info("failed to ensure DNS is enabled on VPC, %v", err)
```

Comment:
This cannot just be an info message if it will result in a Windows VM that cannot be used as a worker node. You need to log an error message here and return an error. Also note that if there are error scenarios that require clean up then you need to do that.